pkg/registry/generic/etcd/etcd.go

```diff
@@ -440,6 +441,33 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	return e.finalizeDelete(out, true)
 }
 
+func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, listOptions *unversioned.ListOptions) (runtime.Object, error) {
```

Comment:
I am wondering if the response should be the following:

1. all the items that were in the listing scope (as i see it is now)
2. all the items that were actually deleted
3. all the items that did not delete 

Since we return immediately on error, I guess (3) is not really possible, and if that is the case, (2) and (1) become the same result.